### PR TITLE
[Merged by Bors] - chore: changes up to nightly-2024-01-22

### PR DESCRIPTION
**Pull request by kim-em**
This is a PR to `bump/v4.6.0`, incorporating the changes on `nightly-testing` up to `nightly-2024-01-22`.

Note that for now this moves `aesop` to the `nightly-testing` branch. Once leanprover-community/aesop#97 lands we can move this back to `bump/v4.6.0`. This PR doesn't need to wait on that, however.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/Rewrites.lean [hunk 1]
```diff
@@ -32,10 +32,10 @@ namespace Lean.Meta
 
 /-- Extract the lemma, with arguments, that was used to produce a `RewriteResult`. -/
 -- This assumes that `r.eqProof` was constructed as:
--- `mkAppN (mkConst ``Eq.ndrec [u1, u2]) #[α, a, motive, h₁, b, h₂]`
--- and we want `h₂`.
+-- `mkApp6 (.const ``congrArg _) α eType lhs rhs motive heq`
+-- in `Lean.Meta.Tactic.Rewrite` and we want `heq`.
```
Comment (semorrison) [inline Mathlib/Tactic/Rewrites.lean:36]: @nomeata's PR leanprover/lean4#3121 changed the proofs that `rw` constructs.

**Review by jcommelin — APPROVED** [review]
Thanks :tada: 

If CI passes, please remove the label `awaiting-CI` and merge this yourself, by adding a comment `bors r+`.

bors d+

**Comment by semorrison** [comment]
bors merge

**Comment by semorrison** [comment]
This broke because Aesop's `nightly-testing` branch was rebased, and the commit we were depending on here vanished. I've switched Aesop back to `bump/v4.6.0` now that it has been updated, and will send to bors again.

**Comment by semorrison** [comment]
bors p=10

bors merge

**Pull request closed without merging**